crypto/cryptlib.c

```diff
@@ -401,8 +401,11 @@ void *OPENSSL_stderr(void)	{ return stderr; }
 int CRYPTO_memcmp(const void *in_a, const void *in_b, size_t len)
 	{
 	size_t i;
-	const unsigned char *a = in_a;
-	const unsigned char *b = in_b;
+	/* Must use "pointers to volatile" to tell the compiler to never
+	 * attempt optimizing the reads. Otherwise the compiler is allowed
+	 * to use LTO and convert this function into canonical version. */
+	const volatile unsigned char *a = in_a;
```

Comment:
@Dmitry-Me Yes, it makes it slower.  That's kinda the point.  It makes it slower still than we'd like, but I don't see how to fix this in a C-coded `memcmp_s()`.  As to point (2), I see that the spec makes accesses to volatile objects through non-volatile pointers undefined, but not the opposite, and it makes no reference to casts, so my reading is that the cast to a `volatile`-qualified type has to be honored.